openupgrade_scripts/scripts/base/16.0.1.3/pre-migration.py

```diff
@@ -12,6 +12,22 @@
 _logger = logging.getLogger(__name__)
 
 
+def enable_coupon_sharing_within_entity(cr):
+    """The coupon_commercial_partner_applicability module is merged into
+    loyalty_partner_applicability. It is checked if it was installed in v15
+    to set the configuration parameter to True to keep the same functionality."""
+    # Check if the module is installed and its status is "installed".
+    if openupgrade.is_module_installed(cr, "coupon_commercial_partner_applicability"):
+        # The value of the configuration parameter is set to True.
+        openupgrade.logged_query(
```

Comment:
Better to use `env["ir.config_parameter"].sudo().set_param(...)` to avoid possible conflicts.

Reply:
But in pre-migration in the base module we don't have "env" available. And I'm not sure if in this case the parameter "config[" "]" could be used.

Reply:
Yes, you have `env` in both pre and post. What you don't have in pre is access to models defined in the module, but it's not the case: `env["ir.config_parameter"]` is accessible except in `base` pre-migration.

Reply:
Oh, sorry, I didn't see that it's `base` pre-migration.